lib/shell/cmd.go

```diff
@@ -17,20 +17,48 @@ package shell
 import (
 	"fmt"
 	"io"
+	"os"
 	"os/exec"
+	"strings"
 	"syscall"
+
+	"github.com/uber/makisu/lib/utils"
 )
 
 // ShellStreamBufferSize is the size of the output buffers when streaming command stdout and stderr
 const ShellStreamBufferSize = 1 << 20
 
-// ExecCommand exec a command given workingDir, cmd and args, returns error if cmd fails
-func ExecCommand(outStream, errStream func(string, ...interface{}), workingDir, cmdName string, cmdArgs ...string) error {
+// ExecCommand exec a cmd and args inside workingDir as user, returns error if cmd fails
+func ExecCommand(outStream, errStream func(string, ...interface{}), workingDir, user, cmdName string, cmdArgs ...string) error {
 	cmd := exec.Command(cmdName, cmdArgs...)
 	if workingDir != "" {
 		cmd.Dir = workingDir
 	}
 
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	currentEnv := os.Environ()
+
+	if user != "" {
```

Comment:
This functions getting real long, can you split this out into a function of its own? It also does 2 things:
1) Finds the syscall.Credential
2) Creates the HOME env variable
We could split those out as well.